lib/Browser.js

```diff
@@ -16,18 +16,69 @@
 
 const helper = require('./helper');
 const Page = require('./Page');
+const EventEmitter = require('events');
 
-class Browser {
+class Browser extends EventEmitter {
   /**
    * @param {!Connection} connection
    * @param {boolean} ignoreHTTPSErrors
    * @param {function()=} closeCallback
    */
   constructor(connection, ignoreHTTPSErrors, closeCallback) {
+    super();
     this._ignoreHTTPSErrors = ignoreHTTPSErrors;
     this._screenshotTaskQueue = new TaskQueue();
     this._connection = connection;
     this._closeCallback = closeCallback || new Function();
+    /** @type {Map<string, Promise<Page>>} */
+    this._pages = new Map();
+    this._connection.on('Target.targetCreated', this._targetCreated.bind(this));
+    this._connection.on('Target.targetDestroyed', this._targetDestroyed.bind(this));
+  }
+
+  /**
+   * @param {!Connection} connection
+   * @param {boolean} ignoreHTTPSErrors
+   * @param {function()=} closeCallback
+   */
+  static async create(connection, ignoreHTTPSErrors, closeCallback) {
+    const browser =  new Browser(connection, ignoreHTTPSErrors, closeCallback);
+    await connection.send('Target.setDiscoverTargets', {discover: true});
+    return browser;
+  }
+
+  /**
+   * @param {{targetInfo: {targetId: string}}} param0
+   */
+  async _targetCreated({targetInfo}) {
+    if (targetInfo.type !== 'page' && targetInfo.type !== 'other')
```

Comment:
The newtab pages come in as 'other'. I'm not sure whether this is intentional or not, but it makes them hard to work with.

Reply:
Please file an upstream bug instead of this.